### PR TITLE
mem-ruby: Fix replacement policy in GPU_VIPER

### PR DESCRIPTION
**Pull request by yuxiaojia**
The current GPU_VIPER protocol's TCC cache update the MRU information
twice with calling a_allocateBlock and ut_updateTag which affects the
LIP and RRIP replacement polies. Remove ut_updateTag fixes the LIP and
RRIP replacement polies.

Change-Id: I79ad9392593e00425a7fe8828048465b2c2c2e1f

### CODE REVIEW

**Review by mattsinc — APPROVED** [review]
Thanks again!  This makes sense to me, but I'll let others chime in before merging.

File: src/mem/ruby/protocol/GPU_VIPER-TCC.sm
Comment (powerjg) [inline src/mem/ruby/protocol/GPU_VIPER-TCC.sm:1162]: Does this still need to do a tag array write?

**Review by powerjg — APPROVED** [review]
lgtm. I will go ahead and approve because I think the answer to my question is "yes"

File: src/mem/ruby/protocol/GPU_VIPER-TCC.sm
Comment (mattsinc) [reply src/mem/ruby/protocol/GPU_VIPER-TCC.sm:1162]: In my opinion yes -- because a_allocateBlock is updating the tag.

**Review by abmerop — APPROVED** [review]

**Comment by mattsinc** [comment]
Thanks again, @yuxiaojia !

**Pull request merged**
